Mathlib/Data/ENNReal/Basic.lean

```diff
@@ -158,7 +157,7 @@ instance : Inhabited ℝ≥0∞ := ⟨0⟩
 /-- Coercion from `ℝ≥0` to `ℝ≥0∞`. -/
 @[coe, match_pattern] def ofNNReal : ℝ≥0 → ℝ≥0∞ := WithTop.some
 
-instance : Coe ℝ≥0 ℝ≥0∞ := ⟨ofNNReal⟩
+instance (priority := high) : CoeTC ℝ≥0 ℝ≥0∞ := ⟨ofNNReal⟩
```

Comment:
If `ENNReal` is an `abbrev`, then why do we need a separate `ofNNReal`? Why not use `WithTop.some`?

Reply:
The abbrev ensures the coercion has a type that makes dot notation work. We could make `ofNNReal` an abbrev too.